daemon/src/main/java/bisq/daemon/grpc/GrpcTradesService.java

```diff
@@ -257,8 +290,55 @@ private GetTradeReply buildGetTradeReply(BsqSwapTrade bsqSwapTrade) {
     private GetTradeReply buildGetTradeReply(Trade trade) {
         boolean wasMyOffer = wasMyOffer(trade);
         String role = getMyRole(trade);
+        var closingStatus = trade.isCompleted()
+                ? coreApi.getClosedTradeStateAsString(trade)
+                : "Pending";
         return GetTradeReply.newBuilder()
-                .setTrade(toTradeInfo(trade, role, wasMyOffer).toProtoMessage())
+                .setTrade(toTradeInfo(trade,
+                        role,
+                        wasMyOffer,
+                        closingStatus).toProtoMessage())
+                .build();
+    }
+
+
+    private GetTradesReply buildGetTradesReply(List<TradeModel> trades, GetTradesRequest.Category category) {
+        // Build trade history list, starting with closed BsqSwap and v1 trades.
+        List<TradeInfo> result = trades.stream()
+                .map(tradeModel -> {
+                    var role = coreApi.getTradeRole(tradeModel);
+                    var isMyOffer = coreApi.isMyOffer(tradeModel.getOffer());
+                    var isBsqSwapTrade = tradeModel instanceof BsqSwapTrade;
+                    var numConfirmations = isBsqSwapTrade
+                            ? coreApi.getTransactionConfirmations(((BsqSwapTrade) tradeModel).getTxId())
+                            : 0;
+                    var closingStatus = category.equals(OPEN)
+                            ? "Pending"
+                            : coreApi.getClosedTradeStateAsString(tradeModel);
+                    return isBsqSwapTrade
+                            ? toTradeInfo((BsqSwapTrade) tradeModel, role, isMyOffer, numConfirmations, closingStatus)
+                            : toTradeInfo(tradeModel, role, isMyOffer, closingStatus);
+                })
+                .collect(Collectors.toList());
+
+        // Add canceled OpenOffers to returned closed trades list.
+        Optional<List<OpenOffer>> canceledOpenOffers = category.equals(CLOSED)
```

Comment:
Why canceled open offers are treated as closed trades?

Reply:
This is consistent with the UI, which shows  `OpenOffer` instances with state == `CANCELED` in the Trade History view.  The UI categorizes them as "Canceled" trades.